lib/client/mediator-subscribers/list.js

```diff
@@ -22,17 +22,23 @@ module.exports = function subscribeToListTopic(syncDatasetTopics, datasetManager
     parameters = parameters || {};
 
     //Creating the item in the sync store
-    datasetManager.list().then(function(arrayOfDatasetItems) {
-      var listDoneTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.LIST, CONSTANTS.DONE_PREFIX, parameters.topicUid);
-
-      self.mediator.publish(listDoneTopic, arrayOfDatasetItems);
-
-    }).catch(function(error) {
-
-      var listErrorTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.LIST, CONSTANTS.ERROR_PREFIX, parameters.topicUid);
-
-      self.mediator.publish(listErrorTopic, error);
+    datasetManager.$fh.sync.getMetaData(datasetManager.datasetId, function(metadata) {
```

Comment:
Can you add `getMetaData` function to the  `DatasetManager`

This will avoid the long chain